### DISCUSSION TITLE
Questions about new training method for large graph

### DISCUSSION THREAD

**Discussion by HelloWorldLTY** [pyg-team/pytorch_geometric#5379] [General]
Hi, I am trying to train a large graph based on graph neural network. I wonder if the only apporach I can use in pyg is to run some sampling method to transfer the data class into dataset class or not? Thanks a lot.

**Comment by rusty1s** [comment]
You can directly take the data object and input them into a mini-batch sampling techniques such as the `NeighborLoader`. This will compute mini-batches on the fly, no need to do the sampling in a pre-processing step (although this is an option as well).
